integration_test.go

```diff
@@ -0,0 +1,358 @@
+	readFilesAndTest(t, goodPath, binaryRoundTripSkipList, func(t *testing.T, path string) {
+		binaryRoundTrip(t, path)
+	})
+}
+
+func TestTextRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, textRoundTripSkipList, func(t *testing.T, path string) {
+		textRoundTrip(t, path)
+	})
+}
+
+func binaryRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	br := NewReaderBytes(b)
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, br, bw)
+	bw.Finish()
+
+	// Make a text writer from the binary writer
+	r := NewReaderBytes(buf.Bytes())
+	str := strings.Builder{}
+	w := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, w)
+	w.Finish()
+
+	// Make another binary writer using the text writer
+	br2 := NewReaderStr(str.String())
+	buf2 := bytes.Buffer{}
+	bw2 := NewBinaryWriter(&buf2)
+	writeToWriterFromReader(t, br2, bw2)
+	bw2.Finish()
+
+	// Compare the 2 binary writers
+	if !reflect.DeepEqual(bw, bw2) {
+		t.Errorf("Round trip test failed on: " + fp)
+	}
+}
+
+func textRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	r := NewReaderBytes(b)
+	str := strings.Builder{}
+	w := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, w)
+	w.Finish()
+
+	// Make a text writer from the binary writer
+	br := NewReaderStr(str.String())
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, br, bw)
+	bw.Finish()
+
+	// Make another binary writer using the text writer
+	r2 := NewReaderBytes(buf.Bytes())
+	str2 := strings.Builder{}
+	w2 := NewTextWriter(&str2)
+	writeToWriterFromReader(t, r2, w2)
+	w2.Finish()
+
+	//compare the 2 text writers
+	if !reflect.DeepEqual(w, w2) {
+		t.Errorf("Round trip test failed on: " + fp)
+	}
+}
+
+func readFilesAndTest(t *testing.T, path string, skipList []string, tf testingFunc) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, file := range files {
+		fp := filepath.Join(path, file.Name())
+		if file.IsDir() {
+			readFilesAndTest(t, fp, skipList, tf)
+		} else if skipFile(skipList, file.Name()) {
+			continue
+		} else {
+			t.Run(fp, func(t *testing.T) {
+				tf(t, fp)
+			})
+		}
+	}
+}
+
+func loadFile(t *testing.T, path string) []byte {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return b
+}
+
+func skipFile(skipList []string, fn string) bool {
+	ion, _ := regexp.MatchString(`.ion$`, fn)
+	bin, _ := regexp.MatchString(`.10n$`, fn)
+
+	return !ion && !bin || isInSkipList(skipList, fn)
+}
+
+func isInSkipList(skipList []string, fn string) bool {
+	for _, a := range skipList {
+		if a == fn {
+			return true
+		}
+	}
+	return false
+}
+
+func writeToWriterFromReader(t *testing.T, r Reader, w Writer) {
+	for r.Next() {
+		name := r.FieldName()
+		if name != "" {
+			w.FieldName(name)
+		}
+
+		an := r.Annotations()
+		if len(an) > 0 {
+			w.Annotations(an...)
+		}
+
+		switch r.Type() {
```

Comment:
Do we need to consider a default case and potentially raise an issue here? Maybe this is handled elsewhere.

Reply:
[reader.Type](https://github.com/Armanbqt/ion-go-1/blob/master/reader.go#L78) returns one of the values defined in [type.go](https://github.com/Armanbqt/ion-go-1/blob/master/type.go#L6). So, there won't be a case where we get anything out of these values.